src/S3/PostObjectV4.php

```diff
@@ -54,46 +54,28 @@ public function __construct(
             'method'  => 'POST',
             'enctype' => 'multipart/form-data'
         ];
+
         // setup basic policy documents
-        if (is_null($expiration)) {
-            $expiration = gmdate('Y-m-d\TH:i:s\Z', strtotime('+1 hours'));
-        }
+        $expiration = date (
+            'Y-m-d\TH:i:s\Z',
+            strtotime(gmdate(self::ISO8601_BASIC) . '+' . $expires .' hours')
```

Comment:
I think this is too restrictive. What if somebody needs to use a specific expiration that is not an even number of hours from the current time? If this argument instead just represents a specific time, customers will have a lot more flexibility in what expiration they can specify.

Take a look at [how the SDK serializes timestamps sent to services](https://github.com/aws/aws-sdk-php/blob/master/src/Api/TimestampShape.php#L25) for an example of taking a specific time as a `DateTime`, string, or integer.